apps/tangle-dapp/utils/formatBn.ts

```diff
@@ -29,7 +29,7 @@ const DEFAULT_FORMAT_OPTIONS: FormatOptions = {
   trimTrailingZeroes: true,
 };
 
-// TODO: Break this function down into smaller local functions for improved legibility and modularity, since its logic is getting complex. Consider making it functional instead of modifying the various variables: Return {integerPart, fractionalPart} per transformation/function, so that it can be easily chainable monad-style.
+// TODO: Break this function down into smaller local functions for improved legibility and modularity, since its logic is getting complex. Consider making it functional instead of modifying the various variables: Return {integerPart, fractionalPart} per transformation/function, so that it can be easily chainable monad-style. Also, prefer usage of Decimal.js instead of BN for better decimal handling without needing to manually handle the edge cases.
```

Comment:
Why not use the `formatBalance` function from PolkadotJS to address these concerns? If you're worried about technical debt, we could replicate its logic sufficiently to suit our needs, simplifying it as necessary.

Reply:
There were some reasons why I decided to use a custom formatting function, I think it was some limitations with the `formatBalance` function, like max decimal place behavior and other things. I'll look into the exact reasons briefly and let you know

Reply:
I believe the reasons were that the `formatBalance` function didn't allow for adding commas, and also on the custom impl. there is better control over the max decimal lengths (ie. some strange behavior if the amount is really small and with the `forceUnit` parameter, where amounts that are not zero are actually returned as `0` which is misleading.)